server/monitor.go

```diff
@@ -2231,6 +2232,11 @@ func (s *Server) Leafz(opts *LeafzOptions) (*Leafz, error) {
 			leafnodes = append(leafnodes, lni)
 		}
 	}
+
+	if len(leafnodes) == 0 {
```

Comment:
Why do we need this?

Reply:
Because it was putting nulls into the json data and the test was poor and didn’t detect this. 

Checking for nil would probably be better but from this code it’s equivalent

Reply:
Could we do omitempty?

Reply:
Yes but since the tests specifically checked this key is present I figured just correcting the data type would be safest and smallest impact.

Reply:
How did test work before?

Reply:
It only checked if the key was in the result. 

I made a mistake at first and one of mine was also null but test was green after I added mine to the cases. 

Checked the test and noticed it was a bit weak so I made the test check for nil which then found mine and this one. 

So test was improved which found it.